src/ui/public/capabilities/route_setup.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { get } from 'lodash';
+import chrome from 'ui/chrome';
+import uiRoutes from 'ui/routes';
+import { UICapabilities } from './ui_capabilities';
+
+uiRoutes.addSetupWork((uiCapabilities: UICapabilities, kbnBaseUrl: string, $route: any) => {
+  const route = _.get($route, 'current.$$route') as any;
+  if (!route.requireUICapabilities) {
+    return;
+  }
+
+  if (!get(uiCapabilities, route.requireUICapabilities)) {
+    const url = chrome.addBasePath(`${kbnBaseUrl}#/home`);
+    window.location.href = url;
+  }
+
+  return undefined;
```

Comment:
We shouldn't need to explicitly return anything here, right?

Reply:
We don't anymore, let me remove this. This is a remnant from when I was trying to use `resolveRedirectTo` which didn't do what I wanted it to.